### PR TITLE
feature/hash-cache-keys

### LINKED ISSUE #48: Optimize Cache Keys
Currently cache keys can be extraordinarily long, which might cause an issue if max key length is reached.

To prevent this, I had the idea of generating cache keys the same way they are now, but then using their hash as the key.

This will reduce key size, as well as obfuscate its content. (That would only be important for encrypted data, which it currently is not, but could also be a future consideration.)

I could see this being a quick little project for you. Please let me know your thoughts on the concept (pros/cons), and if you would be interested in it.

### PR DESCRIPTION
**Pull request by sjbronner**
fixes #48

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 /vendor
 composer.lock
+.*.sw?
```
Comment (mikebronner) [inline .gitignore]: This is probably best handled in your global git configuration. It's been considered "best practice" to leave out environment-specific ignores from the project's git-ignore.

File: src/CachedBuilder.php [hunk 2]
```diff
@@ -123,11 +147,8 @@ public function pluck($column, $key = null)
             return parent::pluck($column, $key);
         }
 
-        $cacheKey = $this->makeCacheKey([$column]) . "-pluck_{$column}";
-
-        if ($key) {
-            $cacheKey .= "_{$key}";
-        }
+        $keyDifferentiator = "-pluck_{$column}" . ($key ? "_{$key}" : "");
```
Comment (mikebronner) [inline src/CachedBuilder.php]: More concise:
`$keyDifferentiator = "-pluck_{$column}_" . ($key ?: "");`

File: src/CachedModel.php [hunk 3]
```diff
@@ -55,7 +55,7 @@ public static function all($columns = ['*'])
         $class = get_called_class();
         $instance = new $class;
         $tags = [str_slug(get_called_class())];
-        $key = 'genealabslaravelmodelcachingtestsfixturesauthor';
+        $key = sha1('genealabslaravelmodelcachingtestsfixturesauthor');
```
Comment (mikebronner) [inline src/CachedModel.php]: Call `makeCacheKey()` here, this was bad code on my part.

Comment (sjbronner) [reply src/CachedModel.php]: This won't work in a static function.

Comment (sjbronner) [reply src/CachedModel.php]: I'm confused as to why the function is marked static in the first place. It is called like an object method.

Comment (mikebronner) [reply src/CachedModel.php]: Something like this?
```php
$key = $instance->makeCacheKey();
```

Comment (sjbronner) [reply src/CachedModel.php]: for instance in `tests/Unit/CachedBuilderTest.php` line 44.

**Pull request merged**